test/e2e/_allocation.js

```diff
@@ -0,0 +1,28 @@
+import { t } from 'testcafe'
+
+import { allocationDetailsPage, allocationCriteriaPage, page } from './pages/'
+
+async function createAllocation() {
+  await allocationDetailsPage.fill()
+  await page.submitForm()
+  await allocationCriteriaPage.fill()
+  await page.submitForm()
+  return t
+    .expect(page.getCurrentUrl())
+    .match(/\/allocation\/[\w]{8}(-[\w]{4}){3}-[\w]{12}\/confirmation$/)
+}
+async function triggerValidationOnAllocationDetailsPage() {
+  await page.submitForm()
+}
+
+async function triggerValidationOnAllocationCriteriaPage() {
+  await allocationDetailsPage.fill()
+  await page.submitForm()
+  await page.submitForm()
```

Comment:
Is this double submit intentional?

Reply:
@teneightfive In this case yes, but in general, I had pushed a bunch of incomplete stuff this morning because I needed to change branch, and I sometimes forget to unstash. It wasn't meant to be reviewed, sorry. I appreciate the eagerness, but maybe next time wait till I re-request the review, just not to waste time.